client.go

```diff
@@ -227,8 +237,18 @@ func (c *Client) Post(url string, contentType ContentType, body interface{}, des
 			}
 			return nil
 		}
+	} else if resp.StatusCode >= 400 && resp.StatusCode < 500 {
+		// Capture responses from 4xx errors
+		var badRequest BadRequestError
```

Comment:
why not just shove the response into `dest`?

Reply:
I thought that at first, but`dest` captures the happy path response, which may not share the same struct as your bad request struct (in this case I'm sure it doesn't).

If the API was perfectly consistent and shared the same structure then I might be more amenable to that idea? As it is I'd rather keep errors as errors and `dest` for "Good Data™" so that it's more obvious when something bad has happened.

Reply:
ah, well then it might be nice to have the full errorBody returned back to you too. There's no 100% guarantee that the error body is `type, message` 😞

Reply:
also there might be extra fields that is useful

Reply:
In that case it'll just return the existing behavior right? My concern right now is that we're creeping scope on this dinghy bug. If we wanted to make this consistent for all error types in Spinnaker it'd push out the fix date for dinghy a bit until I can understand how each service is reporting errors.

In theory it's all consistent, but I'd have to validate for myself if this is the case :/

Reply:
🤷‍♂ you're closer to plank recently than I have been. I'll trust your judgment. I'll approve.

as a FYI, plank was created in like 1 day and iteratively added on, so there's some weirdness and cruft. I think it could welcome much much more improvements, love, and care.

Reply:
That said, if we're really concerned about this change we can follow @german-muzquiz's suggestion here:
https://github.com/armory/plank/pull/62#issuecomment-590475762

Reply:
I'm totally for @german-muzquiz improvement +1+1+1